src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx

```diff
@@ -469,6 +527,12 @@ export const EVMHomeAdaptorProvider = ({
       return "not ready";
 
     try {
+      const gasPrice = await fetchGasPrice();
+
+      if (!gasPrice) {
+        throw new Error('Failed to fetch gas price');
+      }
```

Comment:
I guess it will be better to move error handling inside `fetchGasPrice` to avoid publications

Reply:
Yes, there is DRY, but I did this with two things in mind:
1. The `fetchingGas` function does not need to throw an error in case of failure.
2. We need to add an additional check after `fetchingGas` to handle unexpected behavior.